LEGO1/realtime/vector.h

```diff
@@ -163,13 +163,18 @@ class Vector2 {
 		return *this;
 	}
 
-	// SYNTHETIC: BETA10 0x10013460
-	// Vector3::operator[]
+	// There is another candidate for `Vector2::operator[]` at BETA10 0x10010890, which is called from only three
+	// functions in BETA10:
+	// - `Matrix4::FromQuaternion()`
+	// - `Matrix4::ToQuaternion()`
+	// - `UnknownMx4DPointFloat::FUN_100040a0()`
+	// Maybe there is another subclass of `Vector4` involved that has the same VTABLE but a different `operator[]`.
+	// It is also interesting that `Matrix4::operator[]` is located right above at BETA10 0x10010860.
```

Comment:
This is a weird issue - not exactly sure what is going on here. Binary comparison does not help since these functions are all identical. It would not make sense to have `0x10010890` be a `Matrix4::operator[]` since the places that call it clearly assume a `Vector4`. Maybe `operator[]` is overwritten for `Vector4` by the same code?

Reply:
Yeah, the `operator[]` access functions on the vector/matrix classes are a big mystery. I've been struggling with getting them right for a long time. The issue is that there are a large amount of possible implementations, where inserting (or omitting) an `operator[]` in any of the classes in the vector/matrix hierarchies doesn't make any functional difference, but will affect binary generation across the board. It's very hard to pin down.

Reply:
Feel free to close this thread if you're happy with the comment as it is right now. I am faily confident that at least the two `Vector2::operator[]` BETA10 addresses are correct now (up to permutation between them).

Reply:
Addendum: I just experimented a bit more and found that
```C++
class Vector4 {
	// [...]

	float& operator[](int idx) { return m_data[idx]; }

	// FUNCTION: BETA10 0x10010890
	const float& operator[](int idx) const { return m_data[idx]; }
}
```
appears to only cause a lot of entropy changes in LEGO1 (would need some additional verification) and overall improves the situation on BETA10.

Maybe we'll add that in another PR?